h5py/_hl/attrs.py

```diff
@@ -59,7 +58,7 @@ def __getitem__(self, name):
         if is_empty_dataspace(attr):
             return Empty(attr.dtype)
 
-        dtype = readtime_dtype(attr.dtype, [])
```

Comment:
This presumably did something at some time in the past, but in master at present, `readtime_dtype` is a no-op if `len(names) == 0`, so we can skip it here.